Run `yarn` as part of the publish script

Currently you can run into errors like this when running `npm run publish` if new dependencies are introduced:

```
lerna success - @thumbtack/thumbprint-global-css
lerna success - @thumbtack/thumbprint-react
lerna success - @thumbtack/thumbprint-scss
lerna success - @thumbtack/thumbprint-tokens
$ yarn test:lint:md && yarn test:lint:css && yarn test:lint:js
$ remark "packages/*/CHANGELOG.md" --frail --quiet
$ stylelint --ignore-pattern **/*.module.scss **/*.scss
$ eslint . --ignore-path .gitignore --ext .jsx --ext .js

/Users/doconnor/Thumbtack/thumbprint/packages/thumbprint-codemods/src/avatar-import-name/test.js
  2:29  error  Unable to resolve path to module 'jscodeshift'  import/no-unresolved

/Users/doconnor/Thumbtack/thumbprint/packages/thumbprint-codemods/src/button-secondary-to-tertiary/test.js
  2:29  error  Unable to resolve path to module 'jscodeshift'  import/no-unresolved
```

--- a/package.json
+++ b/package.json
@@ -4,7 +4,7 @@
         "start": "echo \"\n👋 New to Thumbprint? Take a look at our 'CONTRIBUTING.md' for tips!\n\" && yarn run bootstrap && cd www && echo \"\nℹ️  Need a hand? Reach out on #design-systems for help.\n\" && yarn start",
         "updated": "lerna updated",
         "bootstrap": "lerna run prepublishOnly",
-        "publish": "git checkout master && git pull && yarn test && yarn build:dist && lerna publish",
+        "publish": "git checkout master && git pull && yarn && yarn test && yarn build:dist && lerna publish",
         "build:docs": "yarn run bootstrap && cd www && yarn build",
         "build:dist": "lerna exec -- rm -rf dist .cache && yarn run bootstrap",
         "format": "prettier \"**/*.{js,jsx,css,scss,json,md,mdx,html}\" --write",

